api/v1alpha1/authorization_types.go

```diff
@@ -0,0 +1,54 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+// Authorization defines the authorization configuration.
+type Authorization struct {
+	// Rules contains all the authorization rules.
+	// If rules contains at least one Allow rule and none of them
+	// matches the action for the request is deny.
+	// If rules contains at least one Deny rule and none of them
+	// matches the action for the request is allow.
+	// If rules contains Allow and Deny,
+	// and the request matches both, `Deny` takes precedence.
+	//
+	// +kubebuilder:validation:MinItems=1
+	Rules []Rule `json:"rules,omitempty"`
+}
+
+// Rule defines the single authorization rule.
+type Rule struct {
+	// ClientSelector contains the client selector configuration.
+	// All selectors are AND'd together and only if all selector are valid
+	// the Action is performed.
+	//
+	ClientSelector ClientSelector `json:"clientSelector,omitempty"`
+
+	// Action defines the action to be taken if the rule matches.
+	Action RuleActionType `json:"action"`
+}
+
+// ClientSelector contains the client selector configuration.
+type ClientSelector struct {
```

Comment:
`ClientSelector` sounds a bit odd , what do you think of `principal`?  I'm not a native speaker, so my feeling on naming may not be correct anyway :-)

```
// Principle defines the configuration for a principle.
type Principle struct {
	ClientCIDRs []string `json:"clientCIDR,omitempty"`

	// Other type of principal in future: jwt claim, client cert, etc.
}
```

Reply:
`principal` doesnt seem to define the attribute well, I agree `clientSelector` is also odd, @ZackButcher had brought up the same thing, wanna come up with a 3rd name :) ?

Reply:
would it work to call it `matches` similar to what's done on `HTTPRoute`?

Reply:
good point, there was a really long debate about this while designing RateLimit, and `clientSelectors` was chosen there to better describe the `match` field (its way to match/select on specific client attributes) , more in https://github.com/envoyproxy/gateway/pull/706